script.rb

```diff
@@ -0,0 +1,93 @@
+module Enumerable
+  def my_each
+    return to_enum(:each) unless block_given?
+
+    arr = self if instance_of?(Array)
+    arr.length.times { |item| yield(arr[item]) }
+    self
+  end
+
+  def my_each_with_index
+    return to_enum(:each_with_index) unless block_given?
+
+    arr = self if instance_of?(Array)
+    arr.length.times { |item| yield(arr[item], item) }
+    self
+  end
+
+  def my_select
+    new_arr = []
+    return to_enum(:select) unless block_given?
+
+    my_each { |m| new_arr << m if yield(m) }
+    new_arr
+  end
+
+  def my_all?
+    condition = true
+    my_each do |item|
+      condition = false if (block_given? && !yield(item)) || (!block_given? && !item)
+    end
+    condition
+  end
+
+  def my_any?
+    condition = false
+    my_each do |item|
+      condition = true if (block_given? && yield(item)) || (!block_given? && item)
+    end
+    condition
+  end
+
+  def my_none?
+    condition = true
+    my_each do |item|
+      condition = false if (block_given? && yield(item)) || (!block_given? && item)
+    end
+    condition
+  end
```

Comment:
#my_none? returns false if the block ever returns true for all elements::range
#my_none? when a class is passed as an argument returns true if none of the collection is a member of such class
#my_none? when a class is passed as an argument returns true if none of the collection is a member of such class::Numeric
#my_none? when a class is passed as an argument returns false if any of the collection is a member of such class
#my_none? when a Regex is passed as an argument returns true if none of the collection matches the Regex
#my_none? when a Regex is passed as an argument returns false if any of the collection matches the Regex
#my_none? when a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern
#my_none? when a pattern other than Regex or a Class is given returns false only if one of the collection matches the pattern